prow/github/client.go

```diff
@@ -4096,3 +4097,40 @@ func (c *client) GetApp() (*App, error) {
 
 	return &app, nil
 }
+
+// GetDirectory uses GitHub repo contents API to retrieve the content of a directory with commit SHA.
+// If commit is empty, it will grab content from repo's default branch, usually master.
+//
+// See https://developer.github.com/v3/repos/contents/#get-contents
+func (c *client) GetDirectory(org, repo, dirpath, commit string) ([]Content, error) {
+	durationLogger := c.log("GetDirectory", org, repo, dirpath, commit)
+	defer durationLogger()
+
+	url := fmt.Sprintf("/repos/%s/%s/contents/%s", org, repo, dirpath)
+	if commit != "" {
+		url = fmt.Sprintf("%s?ref=%s", url, commit)
+	}
```

Comment:
I think this should at least escape the input (https://golang.org/pkg/net/url/#QueryEscape) but to me it looks that a query should be really handled by a lower layer - we'd be passing something that is no longer a "path" into a `path` member and rely on the implementation to handle this. 

So maybe it's time to add a `query` member into the `request`?

Reply:
Thanks. I have escaped the query-string for now.

To handle query-string escaping in lower layer, we would need to support multiple query params. (example use-case: https://github.com/kubernetes/test-infra/blob/master/prow/github/client.go#L2974)

So, shall we take up this task separately ?

Reply:
Ah, I missed this prior art...  I guess it's fine to just escape that for now.